cmd/warm-backend-azure.go

```diff
@@ -49,108 +59,58 @@ func (az *warmBackendAzure) getDest(object string) string {
 	}
-	blobURL := az.serviceURL.NewContainerURL(az.Bucket).NewBlobURL(az.getDest(object))
-	blob, err := blobURL.Download(ctx, opts.startOffset, opts.length, azblob.BlobAccessConditions{}, false, azblob.ClientProvidedKeyOptions{})
+	resp, err := az.clnt.DownloadStream(ctx, az.Bucket, az.getDest(object), &azblob.DownloadStreamOptions{
+		Range: blob.HTTPRange{Offset: opts.startOffset, Count: opts.length},
+	})
 	if err != nil {
-		return nil, azureToObjectError(err, az.Bucket, object)
+		return nil, azureToObjectError(err, az.Bucket, az.getDest(object))
 	}
 
-	rc := blob.Body(azblob.RetryReaderOptions{})
-	return rc, nil
+	return resp.Body, nil
 }
 
 func (az *warmBackendAzure) Remove(ctx context.Context, object string, rv remoteVersionID) error {
-	blob := az.serviceURL.NewContainerURL(az.Bucket).NewBlobURL(az.getDest(object))
-	_, err := blob.Delete(ctx, azblob.DeleteSnapshotsOptionNone, azblob.BlobAccessConditions{})
-	return azureToObjectError(err, az.Bucket, object)
+	_, err := az.clnt.DeleteBlob(ctx, az.Bucket, az.getDest(object), &azblob.DeleteBlobOptions{})
+	return azureToObjectError(err, az.Bucket, az.getDest(object))
 }
 
 func (az *warmBackendAzure) InUse(ctx context.Context) (bool, error) {
-	containerURL := az.serviceURL.NewContainerURL(az.Bucket)
-	resp, err := containerURL.ListBlobsHierarchySegment(ctx, azblob.Marker{}, "/", azblob.ListBlobsSegmentOptions{
-		Prefix:     az.Prefix,
-		MaxResults: int32(1),
+	maxResults := int32(1)
+	pager := az.clnt.NewListBlobsFlatPager(az.Bucket, &azblob.ListBlobsFlatOptions{
+		Prefix:     &az.Prefix,
+		MaxResults: &maxResults,
 	})
-	if err != nil {
-		return false, azureToObjectError(err, az.Bucket, az.Prefix)
-	}
-	if len(resp.Segment.BlobPrefixes) > 0 || len(resp.Segment.BlobItems) > 0 {
-		return true, nil
-	}
-	return false, nil
-}
-
-func newCredentialFromSP(conf madmin.TierAzure) (azblob.Credential, error) {
-	oauthConfig, err := adal.NewOAuthConfig(azure.PublicCloud.ActiveDirectoryEndpoint, conf.SPAuth.TenantID)
-	if err != nil {
-		return nil, err
-	}
-	spt, err := adal.NewServicePrincipalToken(*oauthConfig, conf.SPAuth.ClientID, conf.SPAuth.ClientSecret, azure.PublicCloud.ResourceIdentifiers.Storage)
-	if err != nil {
-		return nil, err
-	}
-
-	// Refresh obtains a fresh token
-	err = spt.Refresh()
-	if err != nil {
-		return nil, err
-	}
-
-	tc := azblob.NewTokenCredential(spt.Token().AccessToken, func(tc azblob.TokenCredential) time.Duration {
-		err := spt.Refresh()
+	found := false
+	for pager.More() {
+		_, err := pager.NextPage(ctx)
 		if err != nil {
-			return 0
+			return false, err
 		}
-		// set the new token value
-		tc.SetToken(spt.Token().AccessToken)
-
-		// get the next token before the current one expires
-		nextRenewal := float64(time.Until(spt.Token().Expires())) * 0.8
-		if nextRenewal <= 0 {
-			nextRenewal = float64(time.Second)
-		}
-
-		return time.Duration(nextRenewal)
-	})
-
-	return tc, nil
+		found = true
+		break
+	}
+	return found, nil
```

Comment:
Maybe use this to get rid of `SA4004`:
```suggestion
	if !pager.More() {
		return false, nil
	}
	_, err := pager.NextPage(ctx)
	if err != nil {
		return false, err
	}
	return true, nil
```